src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/visualizers/RequestViewHTTP.java

```diff
@@ -287,6 +287,9 @@ public static Map<String, String[]> getQueryMap(String query) {
         String[] params = query.split(PARAM_CONCATENATE);
         for (String param : params) {
             String[] paramSplit = param.split("=");
+            if (paramSplit.length == 0) { // Bug Github 5807
```

Comment:
I don't think the Bug reference is needed here, but I would love to see a unit test :)

```java
public class RequestViewHTTPTest extends TestCase {

    @Test
    void testGetQueryMapWithEmptyKeyAndValue() {
        String query = "=";
        Map<String, String[]> params = RequestViewHTTP.getQueryMap(query);
        Assert.assertNotNull(params);
        Assertions.assertEquals(0, params.size());
    }
 ...
```